### ISSUE TITLE
ShipHull Icon and Graphic are swapped

### ISSUE THREAD

**Issue by dbenage-cx** [freeorion/freeorion#610]
`HullType` graphic name is stored as icon, and icon as graphic.

```
diff --git a/UI/ClientUI.cpp b/UI/ClientUI.cpp
index 804de7f..53fc166 100644
--- a/UI/ClientUI.cpp
+++ b/UI/ClientUI.cpp
@@ -311,6 +311,7 @@ boost::shared_ptr<GG::Texture> ClientUI::HullIcon(const std::string& hull_name)
         if (texture_name.empty())
             texture_name = hull->Graphic();
     }
+    std::cout << "hull: " << hull_name << "\n\ticon: " << hull->Icon() << "\n\tgraphic: " << hull->Graphic() << std::endl;
     if (texture_name.empty())
         return ClientUI::GetTexture(ArtDir() / "icons" / "ship_hulls"/ "generic_hull.png", true);
     return ClientUI::GetTexture(ArtDir() / texture_name, true);
```

returns:

```
hull: SH_SMALL_ASTEROID
    icon: hulls_design/small_asteroid_hull.png
    graphic: icons/ship_hulls/small_asteroid_hull_small.png
...
```

GCC 5.3.1
Boost 1.58

**Comment by dbenage-cx** [comment]
Easiest comparison without changes is looking at solar hull in design window, it has a unique pov between icon and graphic.

Works around the issue, though the unmodified order is correct:

```
diff --git a/parse/ShipHullsParser.cpp b/parse/ShipHullsParser.cpp
index 1c66bb3..bad29b6 100644
--- a/parse/ShipHullsParser.cpp
+++ b/parse/ShipHullsParser.cpp
@@ -107,7 +107,7 @@ namespace {
                 >   parse::detail::common_params_parser()       [ _d = _1 ]
                 >   parse::label(Icon_token)    > tok.string    [ _f = _1 ]
                 >   parse::label(Graphic_token) > tok.string
-                    [ insert_hull(_r1, new_<HullType>(_c, _d, _a, _e, _1, _f)) ]
+                    [ insert_hull(_r1, new_<HullType>(_c, _d, _a, _e, _f, _1)) ]  //TODO: graphic and icon are swapped during construction
                 ;

             start
```

**Comment by MatGB** [comment]
I've got this happening, was intending to do a clean compile to see if it was just me (wouldn't be the first time I've had weird stuff). In my case, the images are swapped for premade hulls and hulls I design myself, but _not_ for hulls I load from the saved file. I've been unable to figure that out.

**Comment by dbenage-cx** [comment]
@Vezzra Might bump version due to low-priority/impact.
I would not have noticed this if I wasn't playing around with callgrind and different fonts.  Unless its something simple I over-looked (highly possible), I suspect its a grammar issue (spirit/lex/qi).
From a player POV, unless MatGB disagrees, there is no noticeable performance or functionality impact outside of the scaling ratio in design window.

**Comment by Vezzra** [comment]
Ok, @MatGB, do you want to have this fixed for 0.4.6, or can we leave that as undecided/optional?

**Comment by MatGB** [comment]
I think it's a bug that ought to be fixed, it's recent, last few weeks, but it first happened to me on a branch full of cherry picks I was messing around with so I can't trace where it comes from (beyond me to do so).

It's not game breaking, but it is weird, especially for the large hulls, have a look at, for example, the Solar Hull in the design page, not a good first impression, especially given we've recently added in all this really cool art that we ought to be highlighting in the release notes.

**Comment by geoffthemedio** [comment]
Probably the result of me reorganizing some parser parameters for hulls and parts into new structs, which was needed to limit the number of parameters passed to their constructors when constructing stuff during the parse.

**Comment by Vezzra** [comment]
Fixed by #626
